identity-account-storage/src/error.rs

```diff
@@ -29,6 +32,13 @@ pub enum Error {
   /// [`KeyType`][identity_core::crypto::KeyType].
   #[error("Invalid Private Key: {0}")]
   InvalidPrivateKey(String),
+  /// Caused by attempting to perform Diffie-Hellman key exchange with the wrong key type
+  #[error("Invalid Key Type: {0}")]
+  InvalidKeyType(String),
+  /// Caused by providing bytes that cannot be used as a public key of the
+  /// [`KeyType`][identity_core::crypto::KeyType].
+  #[error("Invalid Public Key: {0}")]
+  InvalidPublicKey(String),
```

Comment:
I think this crate could present cryptography related errors in a better way. I tried removing the `CoreError` and `CryptoError` variants and pretty much all compilation errors were then the result of missing error types related to results of cryptographic operations. Hence I would suggest removing the `CoreError` and `CryptoError` variants and do one of the following things: 
**Option A:** Make an additional error in this crate `CryptographyError` (and also wrap it in the top level error enum) which you map cryptography related errors from `identity_core` and `crypto.rs` to, or 
**Option B:**  Keep the `InvalidKeyType` and `InvalidPublicKey` variants and add a few more variants to cover what was needed from  `CoreError` and `CryptoError`
**Option C:** Make a new error in the `identity_core::crypto` module covering various failures of cryptographic operations and introduce a new variant here that wraps said error. 

See https://github.com/iotaledger/identity.rs/issues/534 for more information on our error guidelines. I think this is a good opportunity to get better alignment with what our error guidelines state.

Reply:
Let me know if following up on my suggestion would be too much extra work for this PR. If that is the case I would be happy to help in a separate branch.